cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java

```diff
@@ -439,6 +440,61 @@ public void testCustomActionDatasetAccess() throws Exception {
 
   @Category(XSlowTests.class)
   @Test
+  public void testWorkflowLocalDatasets() throws Exception {
+    ApplicationManager applicationManager = deployApplication(testSpace, WorkflowAppWithLocalDatasets.class);
+    final WorkflowManager wfmanager = applicationManager.getWorkflowManager("WorkflowWithLocalDatasets");
+    File waitFile = new File(tmpFolder.newFolder() + "/wait.file");
+    File doneFile = new File(tmpFolder.newFolder() + "/done.file");
+
+    // Write some data to the input file
+    File inputDir = tmpFolder.newFolder();
+
+    File inputFile = new File(inputDir.getPath() + "/words.txt");
```

Comment:
Better use the `File(File parent, String path)` constructor: `new File(inputDir, "words.txt")`